### PR TITLE
Purge Deno files, switch to JSR

### PR DESCRIPTION
**Pull request by colinhacks**
Remove `deno/lib` files and build script
Remove `deno test` runner
Remove `deno.land/x` CI 
Add `jsr.json` with name `@zod/zod` (not published yet)
Migrate to `isolatedDeclarations: true`

### CODE REVIEW

File: packages/zod/tests/json.test.ts [hunk 1]
```diff
@@ -3,9 +3,6 @@ import { expect, test } from "vitest";
 
 import * as z from "../src/index";
 
-// @ts-ignore TS2304
-const isDeno = typeof Deno === "object";
-
```
Comment (mmkal) [inline packages/zod/tests/json.test.ts:8]: woohoo it hurt me to write this!

Comment (colinhacks) [reply packages/zod/tests/json.test.ts:8]: Hahaha I understand the pain!

**Comment by jflatow** [comment]
What happened to the jsr package actually?

**Comment by colinhacks** [comment]
I added the `jsr.json` but I've been waiting for v4 to actually publish it.

**Comment by jflatow** [comment]
Now that v4 (beta) is out, do you think you could publish it? I'm excited to make our `deno.json` right

**Pull request merged**